src/app/wizard/step/cluster/component.ts

```diff
@@ -252,13 +252,16 @@ export class ClusterStepComponent extends StepBase implements OnInit, ControlVal
         this._setDefaultCNIVersion();
       });
 
-    this.control(Controls.Konnectivity)
-      .valueChanges.pipe(takeUntil(this._unsubscribe))
-      .subscribe(value => {
-        if (!value && this.form.get(Controls.ProxyMode).value === ProxyMode.ebpf) {
-          this.form.get(Controls.ProxyMode).setValue('');
+    combineLatest([this.control(Controls.ProxyMode).valueChanges, this.control(Controls.CNIPlugin).valueChanges])
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe(([proxyMode, cniPlugin]) => {
+        const konnectivityControl = this.control(Controls.Konnectivity);
+
+        if (proxyMode === ProxyMode.ebpf && cniPlugin === CNIPlugin.Cilium) {
+          konnectivityControl.disable();
+        } else if (konnectivityControl.disabled) {
+          konnectivityControl.enable();
         }
-        this.updateCNIPluginOptions();
```

Comment:
This method call should be brought back because if `Konnectivity` is unchecked then Proxy modes list is not updated and it still contains `epbf` option

Reply:
If ebpf is selected then it is not possible to change the Konnectivity checkbox as it will be disabled